app/views/handlers/thank_you.py

```diff
@@ -45,4 +53,26 @@ def get_context(self):
                 census_type_code = self.CENSUS_TYPE_MAPPINGS[census_type]
                 break
 
-        return build_census_thank_you_context(self.session_data, census_type_code)
+        return build_census_thank_you_context(
+            self.session_data, census_type_code, self.email_form
+        )
+
+    def validate(self):
+        self.is_valid_email_form = self.email_form.validate_on_submit()
+
+    def get_url_safe_serialized_email(self):
+        url_safe_serializer_handler = URLSafeSerializerHelper()
+        return url_safe_serializer_handler.dumps(self.email_form.email.data)
+
+    def handle_post(self):
+        if not self.email_form:
+            raise NotFound
+
+        self.validate()
+
+        if self.is_valid_email_form:
+            self._update_session_data_confirmation_email_sent_to_true()
```

Comment:
I think the code to handle what we do when someone requests a confirmation email should be done in one place (the `ConfirmationEmail` handler). This will become more important when the logic to keep track of the number of times a confirmation email has been requested is added, as someone can keep returning to the thank you screen to do that. 

We could do this by initialising the ConfirmationEmail handler from the thank-you route (once we know they are sending a POST request and confirmation email is enabled).